torch_xla/core/c10d_registration.py

```diff
@@ -0,0 +1,25 @@
+import torch
+import torch_xla.runtime as xr
+import torch_xla.core.xla_model as xm
+from torch.library import Library
+from typing import List, Optional, TypedDict
+
+c10d_lib = Library("_c10d_functional", "IMPL")
```

Comment:
What is this `Library` for? I don't remember needing this when I was playing with `@torch.library.impl` in `torch_xla2`

Reply:
I think they are the same thing, but using different apis

Reply:
I think it just an easy way to say I want to register the op under `torch.ops. _c10d_functional`, I copied from upstream haha

Reply:
Does this work if you comment this line out? I don't think it's necessary in addition to `torch.library.impl`

Reply:
if you check down below I did
```
@torch.library.impl(c10d_lib, "broadcast", "XLA")
```
I guess I can do
```
@torch.library.impl(Library("_c10d_functional", "IMPL"), "broadcast", "XLA")
```
but then this `@torch.library.impl` might be called for multuple ops so I figured it is better to leave it there.

Reply:
You can reference the ops you need directly by name. See my PR: https://github.com/pytorch/xla/pull/7705/files#diff-8e0394772809eec997864870faa616ae1a5123c0776204e34d0036eb94f69750R67

See the documented example (`aten::sin`): https://pytorch.org/docs/stable/library.html#torch.library.impl

Here, it would be something like `"_c10d_functional::broadcast"`

Reply:
updated